server/fleet/hosts.go

```diff
@@ -202,12 +203,21 @@ type Host struct {
 	DeviceMapping *json.RawMessage `json:"device_mapping,omitempty" db:"device_mapping" csv:"-"`
 }
 
-// DisplayName returns ComputerName if it isn't empty or HostName otherwise.
+// DisplayName returns ComputerName if it isn't empty. Otherwise, it returns Hostname if it isn't
+// empty. If Hostname is empty and both HardwareSerial and HardwareModel are not empty, it returns a
+// composite string with HardwareModel and HardwareSerial. If all else fails, it returns an empty
+// string.
 func (h *Host) DisplayName() string {
-	if cn := h.ComputerName; cn != "" {
-		return cn
+	switch {
+	case h.ComputerName != "":
+		return h.ComputerName
+	case h.Hostname != "":
+		return h.Hostname
+	case h.HardwareModel != "" && h.HardwareSerial != "":
+		return fmt.Sprintf("%s (%s)", h.HardwareModel, h.HardwareSerial)
+	default:
```

Comment:
Let's discuss if this makes sense for the default case.

Reply:
I _think_ it does, or at least this was the implicit behavior (if `Hostname` and `ComputerName` were empty, we returned and empty string)